### DISCUSSION TITLE
Не выполнять код действия, если изменение вызвано скриптом

### DISCUSSION THREAD

**Discussion by nshy** [totumonline/totum-ru-issues-and-discussions#583] [Как это сделать?] [answered]
Привет! 

Хочется чтобы код действия срабатывал, если пользователь редактирует ячейку напрямую. В коде действия я проверяю валидность нового значения. С другой стороны значение в ячейке может быть сгенерено и в таком случае, я не хотел бы делать валидацию, т.к. сгенеренное значение точно валидно, а валидация тяжелая операция. Есть какой-то способ это сделать?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Поставить процессную переменную procVar скриптом который ставит значение и проверять ее в изменяемом поле. 

Только при проверке вызывать ее не быстрой переменной а тоже через функцию с дефаултом и дефаулт при вызове поставить "".

Таким образом если скрипт ставит то там будет значение и не проверяем, если пользователь в поле, то будет присвоенный дефаултом пустой и проверяем.

**Comment by nshy** [comment]
Спасибо!

Была мысль попробовать процессные переменные, но не рискнул) Не понятно из документации, что это такое. Верно ли, что при гонке скрипта запущенного пользователем А и ручным изменением пользователем В, во втором случае будет валидация (строки разные).

В какой момент процессная переменная исчезнет?

**Reply by ttm-alexeypolunin** [reply]
Как процесс PHP завершается она удалается. Те это цепочка действий Код дейст изменил табл там запустился код действия который изменил еще где-то и там запустился код-действия... установленная вначале будет существовать до самого посл действия в цепочке и потом удаляется.
